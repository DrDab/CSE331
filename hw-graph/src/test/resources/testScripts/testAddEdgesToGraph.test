# Create graphs w/ 1, 2, 3 nodes.
CreateGraph g1
AddNode g1 ALPHA

CreateGraph g2
AddNode g2 ALPHA
AddNode g2 BRAVO

CreateGraph g3
AddNode g3 ALPHA
AddNode g3 BRAVO
AddNode g3 CHARLIE

# Try adding 1, 2, 3 edges to graph w/ 1 node
AddEdge g1 ALPHA ALPHA aa1
ListChildren g1 ALPHA
AddEdge g1 ALPHA ALPHA aa2
ListChildren g1 ALPHA
AddEdge g1 ALPHA ALPHA aa3
ListChildren g1 ALPHA

# Try adding 1, 2, 3 edges to graph w/ 2 nodes
AddEdge g2 ALPHA A# Create graphs w/ 1, 2, 3 nodes.
CreateGraph g1
AddNode g1 ALPHA

CreateGraph g2
AddNode g2 ALPHA
AddNode g2 BRAVO

CreateGraph g3
AddNode g3 ALPHA
AddNode g3 BRAVO
AddNode g3 CHARLIE

# Try adding 1, 2, 3 edges to graph w/ 1 node
AddEdge g1 ALPHA ALPHA aa1
ListChildren g1 ALPHA
AddEdge g1 ALPHA ALPHA aa2
ListChildren g1 ALPHA
AddEdge g1 ALPHA ALPHA aa3
ListChildren g1 ALPHA

# Try adding 1, 2, 3 edges to graph w/ 2 nodes
AddEdge g2 ALPHA ALPHA aa1
ListChildren g2 ALPHA
AddEdge g2 ALPHA BRAVO ab1
ListChildren g2 ALPHA
AddEdge g2 BRAVO ALPHA ba1
ListChildren g2 ALPHA
ListChildren g2 BRAVO

# Try adding 1, 2, 3 edges to graph w/ 3 nodes
AddEdge g3 ALPHA ALPHA aa1
ListChildren g3 ALPHA
AddEdge g3 ALPHA BRAVO ab1
ListChildren g3 ALPHA
AddEdge g3 BRAVO CHARLIE bc1
ListChildren g3 ALPHA
ListChildren g3 BRAVO