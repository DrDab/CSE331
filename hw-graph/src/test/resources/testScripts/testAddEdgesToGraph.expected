# Create graphs w/ 1, 2, 3 nodes.
created graph g1
added node ALPHA to g1

created graph g2
added node ALPHA to g2
added node BRAVO to g2

created graph g3
added node ALPHA to g3
added node BRAVO to g3
added node CHARLIE to g3

# Try adding 1, 2, 3 edges to graph w/ 1 node
added edge aa1 from ALPHA to ALPHA in g1
the children of ALPHA in g1 are: ALPHA(aa1)
added edge aa2 from ALPHA to ALPHA in g1
the children of ALPHA in g1 are: ALPHA(aa1) ALPHA(aa2)
added edge aa3 from ALPHA to ALPHA in g1
the children of ALPHA in g1 are: ALPHA(aa1) ALPHA(aa2) ALPHA(aa3)

# Try adding 1, 2, 3 edges to graph w/ 2 nodes
added edge aa1 from ALPHA to ALPHA in g2
the children of ALPHA in g2 are: ALPHA(aa1)
added edge ab1 from ALPHA to BRAVO in g2
the children of ALPHA in g2 are: ALPHA(aa1) BRAVO(ab1)
added edge ba1 from BRAVO to ALPHA in g2
the children of ALPHA in g2 are: ALPHA(aa1) BRAVO(ab1)
the children of BRAVO in g2 are: ALPHA(ba1)

# Try adding 1, 2, 3 edges to graph w/ 3 nodes
added edge aa1 from ALPHA to ALPHA in g3
the children of ALPHA in g3 are: ALPHA(aa1)
added edge ab1 from ALPHA to BRAVO in g3
the children of ALPHA in g3 are: ALPHA(aa1) BRAVO(ab1)
added edge bc1 from BRAVO to CHARLIE in g3
the children of ALPHA in g3 are: ALPHA(aa1) BRAVO(ab1)
the children of BRAVO in g3 are: CHARLIE(bc1)